trs_cli/client.py

```diff
@@ -88,6 +94,81 @@ def __init__(
     #
     # Check DRS-cli repo for examples
 
+    def get_descriptor(
+        self,
+        type: str,
+        id: str,
+        version_id: str,
+        token: Optional[str] = None
+    ) -> Union[FileWrapper, Error]:
+        """Get the tool descriptor for the specified tool.
+
+        Arguments:
+            type: The output type of the descriptor. Plain types return
+                the bare descriptor while the "non-plain" types return a
+                descriptor wrapped with metadata. Allowable values include
+                "CWL", "WDL", "NFL", "GALAXY", "PLAIN_CWL", "PLAIN_WDL",
+                "PLAIN_NFL", "PLAIN_GALAXY".
+            id: A unique identifier of the tool, scoped to this registry.
+            version_id: An identifier of the tool version, scoped to this
+            registry.
+            token: Bearer token for authentication. Set if required by TRS
+                implementation and if not provided when instatiating client or
+                if expired.
+
+        Returns:
+            Unmarshalled TRS response as either an instance of
+            `FileWrapper` in case of a `200` response, or an instance of
+            `Error` for all other JSON reponses.
+
+        Raises:
+            requests.exceptions.ConnectionError: A connection to the provided
+                DRS instance could not be established.
+            trs_cli.errors.InvalidResponseError: The response could not be
+                validated against the API schema.
+        """
+        url = f"{self.uri}/tools/{id}/versions/{version_id}/{type}/descriptor"
```

Comment:
Similar to DRS-cli, the user should be able to provide a TRS URI instead of an `id`. So the first call here should be something like:

```py
id, version_id = self._get_tool_id_version_id(tool_id=id, version_id=version_id)
```